pkg/util/status_check.go

```diff
@@ -137,7 +137,8 @@ func IsAPIServerDeploymentReady(log logr.Logger, c client.Client, hcp tenancyv1a
 	switch hcp.Spec.Type {
 
 	case tenancyv1alpha1.ControlPlaneTypeHost:
-		// host is always available
+	case tenancyv1alpha1.ControlPlaneTypeExternal:
+		// host or external is always available
```

Comment:
This line is only reached when `hcp.Spec.Type` is `tenancyv1alpha1.ControlPlaneTypeExternal`. When `hcp.Spec.Type`, the previous clause (for `tenancyv1alpha1.ControlPlaneTypeHost`) is reached, and now does nothing (whereas it formerly did a `return true, nil`). I suspect that you meant instead of two clauses to have one clause with two matching expressions.